src/examples/go_monorepo_workflows.yml

```
@@ -0,0 +1,180 @@
+description: |
+  This example shows:
+   - Specific version of the isopod.
+   - this is for go application
+   - custom quality checks
+usage:
+  version: 2.1
+
+  orbs:
+    go: circleci/go@1.7.1
+    ric-orb: ricardo/ric-orb@2
+
+  #FIXME: isopod version that you want to use
+  cfg-isopod-version: &cfg-isopod-version
+    isopod_version: 0.29.4
+  #FIXME: go executor version to be used (from circleci's go-orb)
+  cfg-go-build-test-executor: &cfg-go-build-test-executor
+    name: go/default
+    tag: '1.17'
+
+  #FIXME: change with working dirs of your apps
+  cfg-work-dir-worker: &cfg-work-dir-worker './worker'
+  cfg-work-dir-api: &cfg-work-dir-api './api'
+
+  #FIXME: name of the job that github check is looking for
+  build-test-api: &build-test-api 'build-test-api'
+  build-test-worker: &build-test-worker 'build-test-worker'
+
+  parameters:
+    api:
+      type: boolean
+      default: false
+    worker:
+      type: boolean
+      default: false
+    common:
+      type: boolean
+      default: false
+
+  #below there are two workflows, api uses 5 jobs and worker uses 7 jobs to get same thing done
+  #the difference comes from the fact that api uses same docker image for master branch and other branches
+  #while worker builds one image for branch, and one for master, only difference is context.
+  #If you need different context for any reason use worker's approach
+  workflows:
+    version: 2
+    api:
+      when:
+        or:
+          - << pipeline.parameters.api >>
+          - << pipeline.parameters.common >>
+      jobs:
+        - ric-orb/go_build_test:
+            work_dir: *cfg-work-dir-api
+            executor: *cfg-go-build-test-executor
+            #run additional tests on common package
+            additional_tests:
+              - run:
+                name: Running tests on common
+                command: make test
+                working_directory: common
+            name: *build-test-api
```

you might have problems if you have no context at all
```suggestion
        - ric-orb/go_build_test:
            context: dev
            work_dir: *cfg-work-dir-api
            executor: *cfg-go-build-test-executor
            #run additional tests on common package
            additional_tests:
              - run:
                name: Running tests on common
                command: make test
                working_directory: common
            name: *build-test-api
```